src/Stratis.Bitcoin.Features.Consensus/PowConsensusValidator.cs

```diff
@@ -57,6 +69,12 @@ public virtual void CheckBlockHeader(ContextInformation context)
             context.NextWorkRequired = context.BlockValidationContext.ChainedBlock.GetWorkRequired(context.Consensus);
         }
 
+        /// <summary>
+        /// Context-dependent validity checks.
+        /// </summary>
+        /// <param name="context">The context.</param>
```

Comment:
i was actually waiting for this summary and param to see what is this context, how are those checks different - there is 
and i'm disappointed to see that i still don't know what this context is 

can we find out? how are those check methods different, why is this context stuff separated, what is this context? do you know? i don't know.

Reply:
Now when I thought about it I actually can't come up with a good description of this context. It contains basically everything for validation and execution of the block. 

So ho about this: 
"Context that contains variety of information regarding blocks validation and execution."

Reply:
sure, anything more than just "the context" is better